### ISSUE TITLE
Seeding ARRAY(ENUM)

### ISSUE THREAD

**Issue by huntonas** [sequelize/sequelize#11541]
## Issue Description
I am trying to seed an ARRAY(ENUM), but I have not been able to figure out how to do so. I also haven't been able to find any other issue/documentation that showed how to do this.

I keep getting this error:
`ERROR: column "roles" is of type "enum_Users_roles"[] but expression is of type text[]`

This is my migration script:

```
return queryInterface.createTable('Users', {
     id: {
        allowNull: false,
        autoIncrement: true,
        primaryKey: true,
        type: Sequelize.INTEGER
      },  

      email: {
        type: Sequelize.STRING,
	      allowNull: false
      },  

      password: {
        type: Sequelize.STRING,
	      allowNull: false
      },  

      roles: {
        type: Sequelize.ARRAY(Sequelize.ENUM({
	        values: ['user', 'setter', 'admin']
        })),
	      allowNull: false
      },  

      public_id: {
        type: Sequelize.UUID,
	      defaultValue: Sequelize.UUIDV4,
	      allowNull: false
      },  

      createdAt: {
        allowNull: false,
        type: Sequelize.DATE
      },  

      updatedAt: {
        allowNull: false,
        type: Sequelize.DATE
      }  
    })
```
And here is my seed file:
```
'use strict';
const faker = require('faker');

module.exports = {
  up: async (queryInterface, Sequelize) => {
	  return queryInterface.bulkInsert('Users', [
		  {
			  email: faker.internet.email(),
			  roles: ['user'],
			  password: "hash",
			  public_id: faker.random.uuid(),
			  created_at: new Date(),
			  updated_at: new Date()
		  }
	  ]);
  },

  down: (queryInterface, Sequelize) => {

    return queryInterface.bulkDelete('Users', null, {});
  }
};
```

### StackOverflow / Slack attempts

I asked on StackOverflow, but I have not gotten any response (it has only been a day)
[https://stackoverflow.com/questions/58326145/sequelize-seeding-arrayenum](https://stackoverflow.com/questions/58326145/sequelize-seeding-arrayenum)

### Is this issue dialect-specific?

- [ ] No. This issue is relevant to Sequelize as a whole.
- [ ] Yes. This issue only applies to the following dialect(s): XXX, YYY, ZZZ
- [x] I don't know.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [x] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by kaizoku-kenshin** [comment]
Hey, stuck on the same. Did you find any resolution?

**Comment by huntonas** [comment]
@kaizoku-kenshin nope. Tried just about everything I could think of, but I haven't figured it out yet.

**Comment by sushantdhiman** [comment]
You can use this code

```js
class Item extends Sequelize.Model { }

Item.init({
  name: { type: DataTypes.STRING },
  values: {
    type: DataTypes.ARRAY(DataTypes.ENUM({
      values: ['a', 'b']
    }))
  }
}, {
  sequelize,
  timestamps: true
})

sequelize.sync({ force: true }).then(async () => {
  await sequelize.queryInterface.bulkInsert('Items', [
    {
      name: 'xyz',
      values: sequelize.literal(`ARRAY['a']::"enum_Items_values"[]`),
      createdAt: new Date(),
      updatedAt: new Date()
    }
  ]);
});
```

```sql
Executing (default): INSERT INTO "Items" ("name","values","createdAt","updatedAt") VALUES ('xyz',ARRAY['a']::"enum_Items_values"[],'2019-10-16 16:44:30.953 +00:00','2019-10-16 16:44:30.953 +00:00');
```

**Comment by huntonas** [comment]
@sushantdhiman That worked for me. I really appreciate it.

**Comment by papb** [comment]
I'll keep this open as a docs issue, it would be nice to mention this in the docs somewhere. I'll do it, since I am already working on the docs anyway

**Comment by huntonas** [comment]
@papb sorry. Didn't mean to close it on you. The stackoverflow question got a some views and a few thumbs up, so I guess I am not the only one trying to figure it out. Probably a good call to add it to docs. Thanks

**Comment by papb** [comment]
No problem!

**Comment by AbdelrahmanSadat** [comment]
I'm getting the same issue with an array of dates. Has this been resolved or documented yet?

**Comment by AbdelrahmanSadat** [comment]
> I'm getting the same issue with an array of dates. Has this been resolved or documented yet?

For anyone having a similar issue:

https://stackoverflow.com/questions/67026160/sequelize-postgres-date-array-input-being-read-as-text-array

**Comment by MHabashy10** [comment]
Or a more simpler form

    values: '{a}'
please note the curly brases are inside the quotes.

you could add even more values ex.

    values: '{a,b}'
Hope this helps.

**Comment by ollydixon** [comment]
> You can use this code
> 
> ```js
> class Item extends Sequelize.Model { }
> 
> Item.init({
>   name: { type: DataTypes.STRING },
>   values: {
>     type: DataTypes.ARRAY(DataTypes.ENUM({
>       values: ['a', 'b']
>     }))
>   }
> }, {
>   sequelize,
>   timestamps: true
> })
> 
> sequelize.sync({ force: true }).then(async () => {
>   await sequelize.queryInterface.bulkInsert('Items', [
>     {
>       name: 'xyz',
>       values: sequelize.literal(`ARRAY['a']::"enum_Items_values"[]`),
>       createdAt: new Date(),
>       updatedAt: new Date()
>     }
>   ]);
> });
> ```
> 
> ```sql
> Executing (default): INSERT INTO "Items" ("name","values","createdAt","updatedAt") VALUES ('xyz',ARRAY['a']::"enum_Items_values"[],'2019-10-16 16:44:30.953 +00:00','2019-10-16 16:44:30.953 +00:00');
> ```

You can only start and 'init' server once with this.

``` original: error: type "enum_ingredients_allergyTags" already exists```

Seems it's try to set ENUMS again, any ideas?
